lib/go_generator.py

```diff
@@ -202,32 +216,45 @@ def format_comment(self, doc: DocComment, indent: string):
 
         return doc_string
 
-    def doc_item_to_string(self, doc_item: list[DocItem]):
+    def _doc_item_to_string(self, doc_item: list[DocItem]):
         result = []
         for item in doc_item:
             t = item.type
             if t == "text" or t == "bold" or t == "emphasis":
                 result.append(item.text)
             elif t == "ref" or t == "code":
-                result.append(self.ref_to_string(item.text))
+                result.append(self._ref_to_string(item.text))
             elif t == "see":
                 result.append('See')
             elif t == "list":
                 ul = "\n"
                 for li in item.values:
-                    ul += f" - {self.doc_item_to_string(li)}\n"
+                    ul += f" - {self._doc_item_to_string(li)}\n"
                 ul += "\n"
                 result.append(ul)
             else:
-                print(f"unknown doc item type = {t}, consider adding it to doc_item_to_string")
+                _LOG.warning(
+                    f"unknown doc item type = {t}, consider adding it to _doc_item_to_string")
         return ' '.join(result).replace(" ,", ",").replace(" .", ".")
 
-    def ref_to_string(self, ref_value):
+    def _ref_to_string(self, ref_value):
         """
-        :param ref_value: enum_value or enum_type, e.g. roc_endpoint or ROC_INTERFACE_CONSOLIDATED
+        :param ref_value: enum_value or enum_type, e.g. roc_endpoint or ROC_INTERFACE_AUDIO_SOURCE
         :return: go value
         """
-        if ref_value.startswith("roc_"):
-            return to_pascal_case(ref_value.removeprefix('roc_'))
+        if ref_value in self._api_root.doc_refs:
+            ref = self._api_root.doc_refs[ref_value]
+
+            if ref.type == "enum_value":
+                return to_pascal_case(ref_value.removeprefix("ROC_"))
```

Comment:
nit:

let's use `ref.name` instead of `ref_value` and in **else** branch as well. The same way like in java generator

or otherwise - use `ref_value` in java generator and delete `name` field from `DocRef` class